### PR TITLE
Onboarding storage for autogeneration

### PR DESCRIPTION
**Pull request by leni-msft**
```
> azure-schema-generator@1.0.0 generate-single /home/leni/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "storage/resource-manager"

[/tmp/schm_azspc] executing: git fsck
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "storage/resource-manager",
  "namespace": "Microsoft.Storage"
}
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/hlsk8ma2ud5 --tag=all-api-versions --api-version=2015-05-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Installing AutoRest extension '@autorest/azureresourceschema' (3.0.77)
   Installed AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[22.17 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2015-05-01-preview/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2015-05-01-preview
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/bw6yr5od2fd --tag=all-api-versions --api-version=2018-03-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
WARNING: Found duplicate definition for variable-named type Microsoft.Storage/storageAccounts/managementPolicies. Skipping definition with path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/managementPolicies/{managementPolicyName}'.
WARNING: Found duplicate definition for variable-named type Microsoft.Storage/storageAccounts/managementPolicies. Using definition with path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/managementPolicies/{managementPolicyName}'.
[74.78 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-03-01-preview/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-03-01-preview
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/managementPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/d15exarisi --tag=all-api-versions --api-version=2015-06-15 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[20.23 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2015-06-15/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2015-06-15
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/xkn26d7pu4d --tag=all-api-versions --api-version=2016-01-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[19.76 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-01-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2016-01-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/4alpi6y3x76 --tag=all-api-versions --api-version=2016-05-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[20.84 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-05-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2016-05-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/6zrblutzz47 --tag=all-api-versions --api-version=2016-12-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[23.9 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-12-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2016-12-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/ide2v3peqyf --tag=all-api-versions --api-version=2017-06-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[24.76 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-06-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2017-06-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/nhthhknkawa --tag=all-api-versions --api-version=2017-10-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[25.24 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-10-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2017-10-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/dfggo9zc44g --tag=all-api-versions --api-version=2018-02-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[84.66 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-02-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-02-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/h1239uvgjp7 --tag=all-api-versions --api-version=2018-07-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[78.91 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-07-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-07-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/q6hg6li2whk --tag=all-api-versions --api-version=2018-11-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[87.35 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-11-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-11-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/managementPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/l69fzlrhfm --tag=all-api-versions --api-version=2019-04-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[83.61 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2019-04-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2019-04-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/fileServices
- storageAccounts/fileServices/shares
- storageAccounts/managementPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6237 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=/tmp/tbvbmpcclgk --tag=all-api-versions --api-version=2019-06-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.19.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6237/node_modules/@autorest/core/dist' (3.0.6237)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[85.4 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2019-06-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2019-06-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/encryptionScopes
- storageAccounts/fileServices
- storageAccounts/fileServices/shares
- storageAccounts/managementPolicies
- storageAccounts/privateEndpointConnections
================================================================================================================================
```

### CODE REVIEW

File: schemas/2015-05-01-preview/Microsoft.Storage.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          "enum": [
+            "Microsoft.Storage/storageAccounts"
+          ]
+        }
+      },
+      "required": [
+        "apiVersion",
+        "location",
+        "name",
+        "properties",
```
Comment (leni-msft) [inline schemas/2015-05-01-preview/Microsoft.Storage.json]: This `properties` is actually not required, according to @blueww

Comment (anthony-c-martin) [reply schemas/2015-05-01-preview/Microsoft.Storage.json]: Can we add a post-processing step to mark it as non-required?

Comment (leni-msft) [reply schemas/2015-05-01-preview/Microsoft.Storage.json]: We'll try this solution

**Comment by hunmix** [comment]
replace with https://github.com/Azure/azure-resource-manager-schemas/pull/973

**Pull request closed without merging**